src/tests/test_pyodide.py

```diff
@@ -450,3 +450,43 @@ def test_restore_state(selenium):
             import c
         """
     )
+
+
+@pytest.mark.skip_refcount_check
+def test_fatal_error(selenium_standalone):
+    assert selenium_standalone.run_js(
+        """
+        try {
+            pyodide.runPython(`
+                from _pyodide_core import trigger_fatal_error
+                def f():
+                    g()
+                def g():
+                    h()
+                def h():
+                    trigger_fatal_error()
+                f()
+            `);
+        } catch(e){
+            return e.toString();
+        }
+        """
+    )
+    assert (
+        selenium_standalone.logs
+        == dedent(
+            """
+            Python initialization complete
+            Pyodide has suffered a fatal error, refresh the page. Please report this to the Pyodide maintainers.
```

Comment:
Great! As I indeed saw a few such failures with no useful traceback included in the past.